src/components/BookPage/AddReview/index.js

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState } from "react";
+import "./addreview.css";
+
+function Form(props) {
+  const [name, setName] = useState("");
+  const [rating, setRating] = useState("");
+  const [review, setReview] = useState("");
+  const [nameError, setNameError] = useState("");
+  const [ratingError, setRatingError] = useState("");
+  const [reviewError, setReviewError] = useState("");
+  const [reviewCharCount, setReviewCharCount] = useState(0);
+  const maxReviewLength = 450;
+
+  function handleReviewChange(event) {
+    const newReview = event.target.value;
+    if (newReview.length <= maxReviewLength) {
+      setReview(newReview);
+      setReviewCharCount(newReview.length);
+    }
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    setNameError("");
+    setRatingError("");
+    setReviewError("");
+
+    let isValid = true;
+
+    if (name.trim() === "") {
+      setNameError("Name field is required");
+      isValid = false;
+    }
+
+    if (rating === "") {
+      setRatingError("Rating field is required");
+      isValid = false;
+    }
+
+    if (review.trim() === "") {
+      setReviewError("Review field is required");
+      isValid = false;
+    }
+
+    if (isValid) {
+      fetch("https://book-swap-api.dev.io-academy.uk/api/reviews", {
+        method: "POST",
+        body: JSON.stringify({
+          name: `${name}`,
+          rating: rating,
+          review: `${review}`,
+          book_id: id,
+        }),
+
+        mode: "cors",
+        headers: {
+          "Content-Type": "application/json",
+          Accept: "application/json",
+        },
+      })
+        .then((res) => res.json())
+        .then((data) => {
+          console.log(data);
+          setName("");
+          setRating("");
+          setReview("");
+          setReviewCharCount(0);
+        });
+    }
+  }
+
+  function changeName(event) {
+    setName(event.target.value);
+  }
+
+  const id = props.id;
```

Comment:
Why is this here? It only gets used in the fetch request to post the review, so move it into the function that uses it